### PR TITLE
refactor(apig/instance): refactor resource codes and related test

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The current parameter list is old school, the latest API documentation support some of the newest parameters, e.g.
- loadbalancer_provider
- ipv6_enable

Some parameter names do not conform to uniform standards and need to be updated, e.g.
- available_zones
- create_time

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor resource codes and related test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccInstance_ -timeout 360m -parallel 4
=== RUN   TestAccInstance_basic
=== PAUSE TestAccInstance_basic
=== RUN   TestAccInstance_egress
=== PAUSE TestAccInstance_egress
=== RUN   TestAccInstance_ingress
=== PAUSE TestAccInstance_ingress
=== CONT  TestAccInstance_basic
=== CONT  TestAccInstance_ingress
=== CONT  TestAccInstance_egress
--- PASS: TestAccInstance_basic (533.10s)
--- PASS: TestAccInstance_egress (612.29s)
--- PASS: TestAccInstance_ingress (616.61s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      616.682s
```

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_instance.go [hunk 1]
```diff
@@ -44,100 +61,166 @@ func ResourceApigInstanceV2() *schema.Resource {
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Status of the dedicated instance.`,
 			},
-			"supported_features": {
-				Type:     schema.TypeList,
-				Computed: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+			// Deprecated arguments
+			"available_zones": {
```
Comment (hmbig2) [inline huaweicloud/services/apig/resource_huaweicloud_apig_instance.go:213]: add deprecated: "XXX" ?
remove the `[schema: Deprecated]`  from description

File: huaweicloud/services/apig/resource_huaweicloud_apig_instance.go [hunk 2]
```diff
@@ -147,261 +230,222 @@ func buildMaintainEndTime(maintainStart string) (string, error) {
 	regex := regexp.MustCompile("^(02|06|10|14|18|22):00:00$")
 	isMatched := regex.MatchString(maintainStart)
 	if !isMatched {
-		return "", fmtp.Errorf("The start-time format of maintenance window is not 'xx:00:00' or " +
+		return "", fmt.Errorf("the start-time format of maintenance window is not 'xx:00:00' or " +
```
Comment (hmbig2) [inline huaweicloud/services/apig/resource_huaweicloud_apig_instance.go]: this value has been validated in the schema, No need to check again

File: huaweicloud/services/apig/resource_huaweicloud_apig_instance.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_instance.go:213]: the Deprecated label has been added to the parameter 'create_time', according to the discussion, arguments are not necessary.

File: huaweicloud/services/apig/resource_huaweicloud_apig_instance.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_instance.go]: OFC, these codes have been removed.

File: huaweicloud/services/apig/resource_huaweicloud_apig_instance.go [hunk 3]
```diff
@@ -44,364 +61,386 @@ func ResourceApigInstanceV2() *schema.Resource {
...
-			"available_zones": {
-				Type:     schema.TypeList,
-				Required: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
+			"availability_zones": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: `schema: Required, The name list of availability zones for the dedicated instance.`,
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/resource_huaweicloud_apig_instance.go]:
```suggestion
				Description: `schema: Required; The name list of availability zones for the dedicated instance.`,
```

please use `;` as the end of extra schema.

**Pull request merged**